### PR TITLE
3.x Add IAM Policy to allow head node to retrieve compute node console output.

### PR DESCRIPTION
**Pull request by davprat**
### Description of changes
* Changes required for retrieving console output from failed static compute nodes.
   * Add IAM Policy to allow head node to retrieve compute node console output.
   * Add permission to parallelcluster-slurm-head-node policy to allow head node to use batch-get-item DynamoDB call.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,7 +27,7 @@ CHANGELOG
 - Add new configuration parameter `DeletionPolicy` for EFS and FSx for Lustre shared storage to support storage retention.
 - Add new configuration section `Scheduling/SlurmSettings/Database` to enable accounting functionality in Slurm.
 - Add support for On-Demand Capacity Reservations and Capacity Reservations Resource Groups.
-- Add new configuration parameter in `Imds/ImdsSettings` to specify the IMDS version to support in a cluster or build image infrastructure. 
+- Add new configuration parameter in `Imds/ImdsSettings` to specify the IMDS version to support in a cluster or build image infrastructure.
```
Comment (chenwany) [inline CHANGELOG.md:58]: This seems not related to the change

Comment (delongmeng) [reply CHANGELOG.md:58]: is the change here just the deletion of the last space character?

Comment (davprat) [reply CHANGELOG.md:58]: Yes, it was due to the pre-commit hook fixing formatting

**Review by delongmeng — COMMENTED** [review]
looks good to me

**Review by chenwany — DISMISSED** [review]

**Review by chenwany — DISMISSED** [review]

**Review by chenwany — APPROVED** [review]
